purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1559,6 +1571,9 @@ class Purchases internal constructor(
         upgradeInfo: UpgradeInfo,
         listener: ProductChangeCallback
     ) {
+        if (storeProduct.type == ProductType.INAPP && LockedFeature.InAppPurchasing.isLocked) {
```

Comment:
maybe we can move this up to the public method? also wondering if we should put it in the deprecated purchasing methods too, since customers could still be using those...

Reply:
or i guess this might be the method that all of those funnel down to, so maybe this is best. ignore me if you've already thought that through

Reply:
Yeah, the public methods get funneled to these 2 methods, so i added the check here to catch all the public methods (which are a few). There is some minimal logic in the public methods but I think that's ok